File: Sample/Sample/Lua/Items/LinkLabel.lua

local function OnLinkClicked(s, e)
	print(string.format("Link Clicked: %s", e.Link))
end

this = Gwen.DockLayout.Create(Gwen.Canvas)

	label = Gwen.LinkLabel.Create(this)
	label.Dock = Gwen.Dock.Top
	label.HoverColor = Gwen.Color.Create(255, 255, 255, 255)
	label.Text = "Link Label (default font)"
	label.Link = "Test Link"
	label.LinkClicked.add(OnLinkClicked)

	label = Gwen.LinkLabel.Create(this)
	label.Dock = Gwen.Dock.Top;
	label.Font = Gwen.Font.Create("Comic Sans MS", 25)
	label.HoverFont = Gwen.Font.Create("Comic Sans MS", 25, Gwen.FontStyle.Underline)
	label.TextColor = Gwen.Color.Create(255, 0, 80, 205)
	label.HoverColor = Gwen.Color.Create(255, 0, 100, 255)
	label.Text = "Custom Font (Comic Sans 25)"
	label.Link = "Custom Font Link"
	label.LinkClicked.add(OnLinkClicked)

AddItem("Non-Interactive", "LinkLabel", this)

File: Library/Portable/Control/LinkLabel.cs
using System;
using MoonSharp.Interpreter;

namespace GwenNetLua.Control
{
	public class LinkLabel : Label
	{
		public static new LinkLabel Create(Internal.ControlBase parent)
		{
			return new LinkLabel(new Gwen.Control.LinkLabel(parent.Target));
		}

		[MoonSharpHidden]
		public static new LinkLabel Create(Gwen.Control.ControlBase control)
		{
			if (control == null)
				return null;
			else
				return new LinkLabel(control);
		}

		[MoonSharpHidden]
		public LinkLabel(Gwen.Control.ControlBase control)
			: base(control)
		{
		}

		public Color HoverColor { get { return new Color(GetTarget<Gwen.Control.LinkLabel>().HoverColor); } set { GetTarget<Gwen.Control.LinkLabel>().HoverColor = value.Target; } }
		public Font HoverFont { get { return new Font(GetTarget<Gwen.Control.LinkLabel>().HoverFont); } set { GetTarget<Gwen.Control.LinkLabel>().HoverFont = value.Target; } }
		public string Link { get { return GetTarget<Gwen.Control.LinkLabel>().Link; } set { GetTarget<Gwen.Control.LinkLabel>().Link = value; } }

		public event Gwen.Control.ControlBase.GwenEventHandler<Gwen.Control.LinkClickedEventArgs> LinkClicked
		{
			add { GetTarget<Gwen.Control.LinkLabel>().LinkClicked += value; }
			remove { GetTarget<Gwen.Control.LinkLabel>().LinkClicked -= value; }
		}
	}
}
